botbot/apps/plugins/runner.py

```diff
@@ -260,32 +296,59 @@ class RealPlugin(RealPluginMixin, fake_plugin_class):
 
+    def run_plugin(self, line, plugin, plugin_slug, func, arg_dict):
+        with statsd.timer(".".join(["plugins", plugin_slug])):
+            # FIXME: This will not have correct timing if go back to
+            # gevent.
+            # Instantiate a plugin specific to this channel
+            channel_plugin = self.setup_plugin_for_channel(plugin.__class__, line)
+            # get the method from the channel-specific plugin
+            new_func = log_on_error(LOG, getattr(channel_plugin, func.__name__))
+
+            if hasattr(self, 'gevent'):
+                grnlt = self.gevent.Greenlet(new_func, line, **arg_dict)
+                grnlt.link_value(channel_plugin.greenlet_respond)
+                grnlt.start()
+            else:
+                channel_plugin.respond(new_func(line, **arg_dict))
+
+
     def check_for_plugin_route_matches(self, line, router):
         """Checks the active plugins' routes and calls functions on matches"""
         # get the active routes for this channel
-        active_slugs = line._active_plugin_slugs.intersection(router.viewkeys())
+        active_slugs = line._active_plugin_slugs.intersection(router.plugins.viewkeys())
         for plugin_slug in active_slugs:
-            for rule, func, plugin in router[plugin_slug]:
-                match = re.match(rule, line.text, re.IGNORECASE)
-                if match:
-                    LOG.info('Match: %s.%s', plugin_slug, func.__name__)
-                    with statsd.timer(".".join(["plugins", plugin_slug])):
-                        # FIXME: This will not have correct timing if go back to
-                        # gevent.
-                        # Instantiate a plugin specific to this channel
-                        channel_plugin = self.setup_plugin_for_channel(
-                            plugin.__class__, line)
-                        # get the method from the channel-specific plugin
-                        new_func = log_on_error(LOG, getattr(channel_plugin,
-                                                             func.__name__))
-                        if hasattr(self, 'gevent'):
-                            grnlt = self.gevent.Greenlet(new_func, line,
-                                                         **match.groupdict())
-                            grnlt.link_value(channel_plugin.greenlet_respond)
-                            grnlt.start()
-                        else:
-                            channel_plugin.respond(new_func(line,
-                                                            **match.groupdict()))
+            if router.name == "commands":
+                for cmd, func, plugin in router.plugins[plugin_slug]:
+                    args = line.text.split(" ")
+                    if args[0] == self.command_prefix + cmd:
+                        del args[0]
+                        LOG.info('Command: %s.%s', plugin_slug, func.__name__)
+                        arg_dict = {"args": args}
```

Comment:
Instead of del args[0] and passing args, cant you just slice it here to args[1:]?